ledger/participant-integration-api/src/main/scala/platform/apiserver/AuthServiceConfig.scala

```diff
@@ -8,16 +8,25 @@ import com.daml.jwt.{
   ECDSAVerifier,
   HMAC256Verifier,
   JwksVerifier,
-  RSA256Verifier,
   JwtTimestampLeeway,
+  RSA256Verifier,
 }
-import com.daml.ledger.api.auth.{AuthService, AuthServiceJWT, AuthServiceWildcard}
+import com.daml.ledger.api.auth.{AuthService, AuthServiceJWT, AuthServiceWildcard, TimestampLeeway}
 
 sealed trait AuthServiceConfig {
   def create(): AuthService
 }
+
 object AuthServiceConfig {
 
+  def toTimestampLeeway(jwtTimestampLeeway: Option[JwtTimestampLeeway]): Option[TimestampLeeway] = {
+    jwtTimestampLeeway match {
+      case Some(JwtTimestampLeeway(_, Some(expiresAt), _, _)) => Some(TimestampLeeway(expiresAt))
+      case Some(JwtTimestampLeeway(Some(default), None, _, _)) => Some(TimestampLeeway(default))
+      case _ => None
```

Comment:
So if both are set, it would not be applied? what is the logic behind it?

Reply:
If both are set then the first case will be selected. `expiresAt` is more specific and gets prioritised over `default`.